### DISCUSSION TITLE
[BUDI-7347] Design Page Keeps Loading and the Official Fix Does Not Work

### DISCUSSION THREAD

**Discussion by frnksmdlkedjnfr** [Budibase/budibase#12367] [General]
**Checklist**

* [X] I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**

* Self
  * Method: <method> **docker single image & docker compose**
  * Budibase Version: <version> **2\.8.1**
  * App Version: <version> 
* Cloud
  * Tenant ID: <tenantId> 

**Describe the bug**
The design page keeps loading infinitely with Google Chrome, but not so with FireFox. The same happens to preview page and published app page.

**To Reproduce**

1. Clean install of Budibase:
   1\.1 On Debian, run `cd / && mkdir data && cd /data` to create a test folder;
   1\.1 Run `git clone https://github.com/Budibase/budibase.git` to clone the whole repo;
   1\.2 run `cd /data/budibase/hosting && docker compose up` to run the docker containers defined in the .yml file.
2. Open Budibase and Create the First App
   2\.1 With Chrome, open \[server-ip\]:10000 to enter the front page, then create an admin account before logging in;
   2\.2 Click the 'Let's go!' button to create the first app;
   2\.3 Click the 'Design' tab.
3. Problem Occurs: The screen preview keeps loading

![](https://uploads.linear.app/7ecfa297-cc77-4fdf-9117-fa5afdb8f908/08b010d9-0ffd-4a3b-9080-0af384c4bf60/13c8e3ad-9df8-4596-8221-036490986f44?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzdlY2ZhMjk3LWNjNzctNGZkZi05MTE3LWZhNWFmZGI4ZjkwOC8wOGIwMTBkOS0wZmZkLTRhM2ItOTA4MC0wYWYzODRjNGJmNjAvMTNjOGUzYWQtOWRmOC00NTk2LTgyMjEtMDM2NDkwOTg2ZjQ0IiwiaWF0IjoxNjkwODk2OTM5LCJleHAiOjE2OTA5ODMzMzl9.TpGxR0rklJ8VguGYq2xKhcIdr9P6rKl24LbhuN5AIic)

**Expected behavior**
The design page shall load instantly as it does on Firefox:

![](https://uploads.linear.app/7ecfa297-cc77-4fdf-9117-fa5afdb8f908/189dde90-ef44-41b4-9a1f-c9b7462da39c/76e6d318-6c87-4fb2-bce8-77467b5df55c?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzdlY2ZhMjk3LWNjNzctNGZkZi05MTE3LWZhNWFmZGI4ZjkwOC8xODlkZGU5MC1lZjQ0LTQxYjQtOWExZi1jOWI3NDYyZGEzOWMvNzZlNmQzMTgtNmM4Ny00ZmIyLWJjZTgtNzc0NjdiNWRmNTVjIiwiaWF0IjoxNjkwODk2OTM5LCJleHAiOjE2OTA5ODMzMzl9.yHq2PBcRtv-13mTsfYRxRqqIx6SgcsxLenEx2sNwZvM)

**App Export**
If possible - please attach an export of your budibase application for debugging/reproduction purposes.

**Desktop (please complete the following information):**

* OS:
  `PRETTY_NAME="Debian GNU/Linux 11 (bullseye)" NAME="Debian GNU/Linux" VERSION_ID="11" VERSION="11 (bullseye)" VERSION_CODENAME=bullseye ID=debian`
* Browser: **Cent Browser**
* Version: **5\.0.1002.354 (Official Build) (64-bit) (Chromium 102.0.5005.167)**

**Additional context**
Installing using the official Docker command `docker run -d -t \   --name=budibase \   -p 10000:80 \   -v /local/path/data:/data \   --restart unless-stopped \   budibase/budibase:latest` yielded the same problem.

**Attempted fixes**
As per the [official docs](https://github.com/Budibase/budibase/blob/develop/docs/DEV-SETUP-DEBIAN.md#file-descriptor-issues-with-vite-and-chrome-in-linux), I have added `* - nofile 65536` to `/etc/security/limits.conf.` both inside the container and the physical Debian device.

Steps:

1. Inside the Container
   1\.1 Run `docker ps` to get the ID of the Budibase container: `ababf50ecf6e`;

![](https://uploads.linear.app/7ecfa297-cc77-4fdf-9117-fa5afdb8f908/6f74ccbb-17f9-42cc-b2c4-ad853b76816a/2cd61165-96c7-4811-a530-95f5e80fdf45?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzdlY2ZhMjk3LWNjNzctNGZkZi05MTE3LWZhNWFmZGI4ZjkwOC82Zjc0Y2NiYi0xN2Y5LTQyY2MtYjJjNC1hZDg1M2I3NjgxNmEvMmNkNjExNjUtOTZjNy00ODExLWE1MzAtOTVmNWU4MGZkZjQ1IiwiaWF0IjoxNjkwODk2OTM5LCJleHAiOjE2OTA5ODMzMzl9.wJwaZ64Gp0SfOLErFrsv7tDTa-J5d1kJfqukXXdSG5Y)

1\.2 Run `docker exec -it ababf50ecf6e sh` to get inside the container;
1\.3 Run `apt update && apt install nano -y && nano /etc/security/limits.conf`;
1\.4 Append `* - nofile 65536` before the EOF line, then save the file.

![](https://uploads.linear.app/7ecfa297-cc77-4fdf-9117-fa5afdb8f908/246593fb-2f8d-4d41-9261-38dd09b1e7ed/34f70e0d-8556-480f-9c2b-edd1ac84229b?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzdlY2ZhMjk3LWNjNzctNGZkZi05MTE3LWZhNWFmZGI4ZjkwOC8yNDY1OTNmYi0yZjhkLTRkNDEtOTI2MS0zOGRkMDliMWU3ZWQvMzRmNzBlMGQtODU1Ni00ODBmLTljMmItZWRkMWFjODQyMjliIiwiaWF0IjoxNjkwODk2OTM5LCJleHAiOjE2OTA5ODMzMzl9.ffUHOIy0UBtGZ0QzSWANkNOOHFYW40wB0GE1BqWmjts)

1. On the Host Machine
   2\.1 Run `apt update && apt install nano -y && nano /etc/security/limits.conf`;
   2\.2 Append `* - nofile 65536` before the EOF line, then save the file.

![](https://uploads.linear.app/7ecfa297-cc77-4fdf-9117-fa5afdb8f908/134f3596-0c5a-4716-aa38-5192eb7b1671/f1d574f2-3ada-4873-9269-2d42b85a7a72?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzdlY2ZhMjk3LWNjNzctNGZkZi05MTE3LWZhNWFmZGI4ZjkwOC8xMzRmMzU5Ni0wYzVhLTQ3MTYtYWEzOC01MTkyZWI3YjE2NzEvZjFkNTc0ZjItM2FkYS00ODczLTkyNjktMmQ0MmI4NWE3YTcyIiwiaWF0IjoxNjkwODk2OTM5LCJleHAiOjE2OTA5ODMzMzl9.8EoGvT_Ov40NytUGxgm7-exh_ofvDaeiIikWSpKTU2I)

1. Reboot the Host Machine
   3\.1 Run `reboot`, then log into the host machine;
   3\.2 Open \[server-ip\]:10000 with Chrome;
   3\.3 Problem persists as described before.

**Screenshots**
Network activities:

![](https://uploads.linear.app/7ecfa297-cc77-4fdf-9117-fa5afdb8f908/dfd3a237-01af-42c4-b7f4-7b466b3d0b36/bc032080-444f-487c-9ce0-071c676fbc74?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzdlY2ZhMjk3LWNjNzctNGZkZi05MTE3LWZhNWFmZGI4ZjkwOC9kZmQzYTIzNy0wMWFmLTQyYzQtYjdmNC03YjQ2NmIzZDBiMzYvYmMwMzIwODAtNDQ0Zi00ODdjLTljZTAtMDcxYzY3NmZiYzc0IiwiaWF0IjoxNjkwODk2OTM5LCJleHAiOjE2OTA5ODMzMzl9.kPmLiULo3hf49KFcNDupkqFQ1sv06IS3XB2oBR3Zzx0)

![](https://uploads.linear.app/7ecfa297-cc77-4fdf-9117-fa5afdb8f908/c00f60f6-5caa-40c9-af06-a49dc837e45c/2dce2e87-3565-4cb8-b9a2-bfaacea3781f?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzdlY2ZhMjk3LWNjNzctNGZkZi05MTE3LWZhNWFmZGI4ZjkwOC9jMDBmNjBmNi01Y2FhLTQwYzktYWYwNi1hNDlkYzgzN2U0NWMvMmRjZTJlODctMzU2NS00Y2I4LWI5YTItYmZhYWNlYTM3ODFmIiwiaWF0IjoxNjkwODk2OTM5LCJleHAiOjE2OTA5ODMzMzl9.WZ9u6Hz1NYT8g7tQ4KEKSbzBDWrHEatpegC7PizL6Hc)

<sub>From [SyncLinear.com](https://synclinear.com) | [BUDI-7347](https://linear.app/budibase/issue/BUDI-7347/design-page-keeps-loading-and-the-official-fix-does-not-work)</sub>

**Comment by aptkingston** [comment]
Hi @frnksmdlkedjnfr. Are you trying to develop with Budibase, or just run it self hosted? This is an unsupported hosting strategy, and the official fix only applies to dev setups.

Please read https://docs.budibase.com/docs/hosting-methods for the official hosting methods.

**Comment by frnksmdlkedjnfr** [comment]
> Hi @frnksmdlkedjnfr. Are you trying to develop with Budibase, or just run it self hosted? This is an unsupported hosting strategy, and the official fix only applies to dev setups.
> 
> Please read https://docs.budibase.com/docs/hosting-methods for the official hosting methods.

I was trying to run BudiBase self-hosted. And why my strategy is unsupported? Aren't I just self-deploying BudiBase with Docker the conventional way?

**Comment by aptkingston** [comment]
Sorry @frnksmdlkedjnfr - I skimmed over this pretty quickly and I think the repo cloning threw me, but you're completely right that that's an entirely valid way to grab the docker-compose config and the environment file.

I'd also like to say thank you for the extremely detailed issue!

The thing that stands out most here is that the same thing happens when you run the "single image" docker container... That rules out the vast majority of docker/hosting related issues, especially around the Minio container, which would normally be where I would point the finger for an error like this.

That leaves me thinking that this is an error specific to your browser, which I notice is Cent Browser.

A few questions which should help us get to the bottom of this:
- Do you have another chromium based browser, or even Chrome itself, which you could try? I don't know of anyone else having tried it with Cent, so it could be that there's an issue with that specific chromium implementation.
- Could you try publishing your app, then viewing the published app? I would like to see if it loads properly when published as this does not depend on message passing across an iframe boundary, which it does in the builder and the preview window. This is where some browser-specific behaviour could cause issues, with different security permissions around iframe origins.

Again, apologies for closing this prematurely.

**Comment by frnksmdlkedjnfr** [comment]
> Sorry @frnksmdlkedjnfr - I skimmed over this pretty quickly and I think the repo cloning threw me, but you're completely right that that's an entirely valid way to grab the docker-compose config and the environment file.
> 
> I'd also like to say thank you for the extremely detailed issue!
> 
> The thing that stands out most here is that the same thing happens when you run the "single image" docker container... That rules out the vast majority of docker/hosting related issues, especially around the Minio container, which would normally be where I would point the finger for an error like this.
> 
> That leaves me thinking that this is an error specific to your browser, which I notice is Cent Browser.
> 
> A few questions which should help us get to the bottom of this:
> 
> * Do you have another chromium based browser, or even Chrome itself, which you could try? I don't know of anyone else having tried it with Cent, so it could be that there's an issue with that specific chromium implementation.
> * Could you try publishing your app, then viewing the published app? I would like to see if it loads properly when published as this does not depend on message passing across an iframe boundary, which it does in the builder and the preview window. This is where some browser-specific behaviour could cause issues, with different security permissions around iframe origins.
> 
> Again, apologies for closing this prematurely.

It's all good. As for your suggestions, below are the results:

1. Other Chromium-based browsers:
I have tried the same clean set-up with Microsoft Edge, Internet Explorer (not Chromium based), and Google Chrome. None of these managed to load the design page. Internet Explorer won't even load anything.
![2023-08-02_16-39-57_Settings - Profile 1 - Microsoft​ Edge](https://github.com/Budibase/budibase/assets/63704404/12c53ea8-fee3-4eff-8585-4c5730861917)
![2023-08-02_16-44-36_Settings - About Chrome - Google Chrome](https://github.com/Budibase/budibase/assets/63704404/fb7383c7-06a5-4133-9008-ef2275ea00f2)
![2023-08-02_16-45-07_About Internet Explorer](https://github.com/Budibase/budibase/assets/63704404/2cced2b0-36da-4835-ab88-5ed324ce8a3f)

2. Publishing the app:
Yields the same results. I actually have tried publishing apps long ago. No browser except Firefox could open the app page.
![2023-08-02_16-47-08_Settings – About Cent Browser - Cent Browser](https://github.com/Budibase/budibase/assets/63704404/97afb9df-3532-4f44-a912-6b0264a5e674)
